### PR TITLE
[DDP-8631] add country field to medical provider suppport

### PR DESCRIPTION
**Pull request by gskhirel**
https://broadinstitute.atlassian.net/jira/software/c/projects/DDP/boards/836?modal=detail&selectedIssue=DDP-8631&assignee=627ecf988dd5f30068573750
1. Medical provider components didn't store country after submitting Medical Release
2. Updated Medical provider Routes and components by adding country field.

### CODE REVIEW

File: pepper-apis/dss-database/src/main/resources/db-changes/schema/DDP-8631-add-country-field-to-medical-provider.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                   http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+
+    <changeSet author="gskhirel" id="physician_country_add">
+        <addColumn tableName="user_medical_provider">
+            <column name="country" type="varchar(100)" />
```
Comment (bskinner) [inline pepper-apis/dss-database/src/main/resources/db-changes/schema/DDP-8631-add-country-field-to-medical-provider.xml]: Let's add a constraint to explicitly make this nullable (Liquibase's default is database-dependent).

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 2]
```diff
@@ -2713,6 +2713,8 @@ MedicalProvider:
       type: string
     state:
       type: string
+    country:
+      type: string
```
Comment (bskinner) [inline pepper-apis/docs/specification/src/components/schemas.yml:2717]: If this is not a required field, it should be marked as `nullable`
```suggestion
    country:
      type: string
      nullable: true
```

**Review by bskinner — COMMENTED** [review]
Couple of minor changes, but looks good otherwise. @MocanaAtBroad @ssettipalli - Anything you'd like to add?

**Comment by bskinner** [comment]
There are a number of [failing tests](https://app.circleci.com/pipelines/github/broadinstitute/ddp-study-server/16096/workflows/9dfcad54-2991-4ad8-a999-d01a78d320f9/jobs/30512/tests) which need to be resolved before this can be merged.

**Comment by gskhirel** [comment]
I think, tests fail because of connection problem with pubsub.

**Comment by bskinner** [comment]
> I think, tests fail because of connection problem with pubsub.

The failing tests are JDBI/SQL related errors, not pubsub connection issues. My guess is there's an SQL statement that hasn't been updated with the changes to the `MedicalProviderDto` constructor.

One of the errors, in test `testConvertSubstitutionToPdfFailsWhenMissingFields_ActivityDate`:
```
java.lang.IllegalArgumentException: Instance factory 'public org.broadinstitute.ddp.db.dto.MedicalProviderDto(java.lang.Long,java.lang.String,long,long,org.broadinstitute.ddp.model.activity.types.InstitutionType,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String)' parameter '[mcountry]' has no matching columns in the result set. Verify that the Java compiler is configured to emit parameter names, that your result set has the columns expected, annotate the parameter names explicitly with @ColumnName, or annotate nullable parameters as @Nullable
	at org.broadinstitute.ddp.service.PdfGenerationServiceTest.lambda$testConvertSubstitutionToPdfFailsWhenMissingFields_ActivityDate$21(PdfGenerationServiceTest.java:398)
	at org.broadinstitute.ddp.service.PdfGenerationServiceTest.testConvertSubstitutionToPdfFailsWhenMissingFields_ActivityDate(PdfGenerationServiceTest.java:398)
```

It appears all the failures are occurring in the `PdfGenerationService.loadParticipantData(Handle,PdfConfiguration,String)` method.

**Comment by bskinner** [comment]
> It appears all the failures are occurring in the `PdfGenerationService.loadParticipantData(Handle,PdfConfiguration,String)` method.

Did some digging- the templated query `bulkQueryLatestEnrollmentsWithUserDataAndOrderedProvidersByStudyId`, used by the method `ParticipantDao._findParticipantsWithUserDataByStudyId(long,boolean,boolean,Set<Long>,Set<Long>)` needs to be updated.

**Review by bskinner — APPROVED** [review]

**Pull request merged**